### PR TITLE
fix sirepo#5041 make slack links work and open in new window

### LINKED ISSUE #5041: Jupyter:  the Slack icon should open a new tab
When clicking on the Slack icon in Sirepo/Jupyter it should open a new tab.

image

The present behavior is confusing, because one loses the Jupyter server session.

### PR DESCRIPTION
**Pull request by rorour**
The code in this repo is broken on prod.
- Make slack links work
- Changed Visit -> Join to be consistent with other Slack link in sirepo menu
- Fix https://github.com/radiasoft/sirepo/issues/5041

note, `pip install -e .` wasn't tracking my changes to lib/index.js, so i kept having to `pip install -e .` every time i made changes. Maybe related to the complicated setup.py?

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -132,6 +132,35 @@
     sanitize-html "~2.5.3"
     url "^0.11.0"
 
+"@jupyterlab/apputils@^3.1.18":
```
Comment (rorour) [inline yarn.lock:135]: I wasn't sure about committing this file, or if it's necessary at all.

File: README.md [hunk 2]
```diff
@@ -49,7 +47,7 @@ Run a sirepo jupyter server from sirepo repo directory
 cd ~/src/radiasoft/sirepo
 ```
 ```bash
-bash etc/run-jupyterhub.sh
+bash etc/run.sh jupyterhub
```
Comment (e-carlin) [inline README.md:50]: This should be fixed lower down in the README too.

File: yarn.lock (hunk 1 above)
Comment (e-carlin) [reply yarn.lock:135]: It's necessary. This is a [lock file](https://stackoverflow.com/a/39992365). It more tightly controls the dependencies (specifies them exactly). By committing it then any other developers (or our build system) are sure to get exactly the dependencies outlined here which are the ones that we tested on.

File: lib/index.js [hunk 3]
```diff
@@ -19,25 +17,22 @@ const logo = new LabIcon({
...
+        this.title.icon = logo;
+        this.node.classList.add('slack-logo');
+    }
 }
 
-
-async function getUrlWithUID(Url) {
-  fetch(Url, {
+function getSlackUrlWithUid() {
+  return fetch('/auth-state', {
```
Comment (e-carlin) [inline lib/index.js:31]: I'd add 
```
// POSIT: sirepo.simulation_db.SCHEMA_COMMON.route.authState
```

Usually when I search for routes I search for their key name (authState) not the uri (auth-state). Adding this we are at least a little closer to catching this use of authState which is outside of the Sirepo app and likely to be missed if we are ever looking for all uses.

File: lib/index.js [hunk 4]
```diff
@@ -68,23 +61,21 @@ export default [
...
-                                </a>
-                        </div>
-                      `;
-      widget.node.appendChild(n);
-      widget.id = 'simple-widget-example';
-      console.log('widget.node: ', widget.node);
-      widget.node.classList.add('slack-logo');
-      shell.add(widget, 'left', {rank: 4});
-
+      var widget = new SlackWidget();
```
Comment (e-carlin) [inline lib/index.js]: Why change to var?

**Review by e-carlin — CHANGES_REQUESTED** [review]
A few nits.

> i kept having to pip install -e . every time i made changes

`jlpm run build` worked for me. Does that work on your machine?

File: README.md (hunk 2 above)
Comment (rorour) [reply README.md:50]: What line?

**Review by gurhar1133 — COMMENTED** [review]
Looks good to me, the link opens in a new tab.
Not sure if it matters on dev, but the link seems to have an issue though:
Screenshot 2024-06-04 at 11 55 49 AM

**Comment by rorour** [comment]
@gurhar1133 thanks! that is just due to the default config value for the slack link in dev- it will work in prod

File: lib/index.js (hunk 3 above)
Comment (rorour) [reply lib/index.js:31]: 46e3f029c06bba1657a5a6253d3a223fcf5da9ca

File: lib/index.js (hunk 4 above)
Comment (rorour) [reply lib/index.js]: I wasn't very familiar with using const for objects in JS. After reading about it I've changed it back. 42fb1f3652efd9e334cd70ed465d4ccf4db14a6e

**Comment by rorour** [comment]
@e-carlin yes, running jlpm run build after every change works for me.

Ready for re-review

File: README.md (hunk 2 above)
Comment (e-carlin) [reply README.md:50]: Sorry, lost in the sauce. Not sure what yesterday evan was thinking.

**Review by e-carlin — APPROVED** [review]

**Comment by e-carlin** [comment]
I'll build beamsim-jupyter-base to release.

**Pull request merged**